app/main.py

```diff
@@ -1,8 +1,24 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.__class__.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people_list: list) -> list:
+    person_list = []
+    for human in people_list:
+        person = Person(human["name"], human["age"])
+        spouse = human.get("wife") or human.get("husband")
+        if spouse in Person.people:
+            spouse_link = Person.people[spouse]
+            if "wife" in human:
```

Comment:
you still doing "double-check" and overall logic is a bit too complicated

Reply:
corrected